### ISSUE TITLE
Why did my installation fail?

### ISSUE THREAD

**Issue by Coolboomer9** [WENO-OF/WENOEXT#54]
-------------------------------------------------------------------
	Build State
-------------------------------------------------------------------
Built state is: unspecified
Building tests
This can be deactivated by using CMAKE_BUILD_TYPE=Release
CMake Error at tests/CMakeLists.txt:2 (add_subdirectory):
  The source directory

    /home/wen/Desktop/WENOEXT/tests/Catch2

  does not contain a CMakeLists.txt file.

-------------------------------------------------------------------

-- Configuring incomplete, errors occurred!
See also "/home/wen/Desktop/WENOEXT/build/CMakeFiles/CMakeOutput.log".

**Comment by Coolboomer9** [comment]
With the above problem solved, the new question is does this format support openfoam6?

**Comment by croquemadame** [comment]
> With the above problem solved, the new question is does this format support openfoam6?

May I ask how you solved the problem above?

**Comment by JanGaertner** [comment]
It is supporting OpenFOAM 5.x to version 8.

**Comment by JanGaertner** [comment]
> ```
> Build State
> ```
> 
> Built state is: unspecified Building tests This can be deactivated by using CMAKE_BUILD_TYPE=Release CMake Error at tests/CMakeLists.txt:2 (add_subdirectory): The source directory
> 
> ```
> /home/wen/Desktop/WENOEXT/tests/Catch2
> ```
> 
> does not contain a CMakeLists.txt file.
> 
> -- Configuring incomplete, errors occurred! See also "/home/wen/Desktop/WENOEXT/build/CMakeFiles/CMakeOutput.log".

Regarding this issue: You probably did not clone the git submodules. I had changed from the single header Catch2 file to the new Catch2 v3 release which is an own library. This is included as a git submodule.

After cloning execute:
```
git submodule init
git submodule update
```

**Comment by croquemadame** [comment]
> > ```
> > Build State
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Built state is: unspecified Building tests This can be deactivated by using CMAKE_BUILD_TYPE=Release CMake Error at tests/CMakeLists.txt:2 (add_subdirectory): The source directory
> > ```
> > /home/wen/Desktop/WENOEXT/tests/Catch2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > does not contain a CMakeLists.txt file.
> > -- Configuring incomplete, errors occurred! See also "/home/wen/Desktop/WENOEXT/build/CMakeFiles/CMakeOutput.log".
> 
> Regarding this issue: You probably did not clone the git submodules. I had changed from the single header Catch2 file to the new Catch2 v3 release which is an own library. This is included as a git submodule.
> 
> After cloning execute:
> 
> ```
> git submodule init
> git submodule update
> ```

Thanks, it worked!

**Comment by Tohid2020** [comment]
Hello Dear All, 
I hope this message finds you well. I have problem in installing WENO. How do I fix the error below?                                                                          [  1%] Building CXX object libWENOEXT/CMakeFiles/WENOEXT.dir/BlazeIO/BlazeIO.C.o
In file included from weno/WENOEXT/blaze-3.8/blaze/system/Platform.h:43,
                 from weno/WENOEXT/blaze-3.8/blaze/util/Limits.h:44,
                 from weno/WENOEXT/blaze-3.8/blaze/math/Accuracy.h:44,
                 from weno/WENOEXT/blaze-3.8/blaze/Math.h:44,
                 from weno/WENOEXT/libWENOEXT/BlazeIO/BlazeIO.H:46,
                 from weno/WENOEXT/libWENOEXT/BlazeIO/BlazeIO.C:29:
weno/WENOEXT/blaze-3.8/blaze/system/CacheSize.h:77:69: error: static assertion failed: Compile time condition violated
   77 | BLAZE_STATIC_ASSERT( blaze::cacheSize > 10000UL && blaze::cacheSize < 2048000UL );
      |                                                    ~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/weno/WENOEXT/blaze-3.8/blaze/util/StaticAssert.h:113:19: note: in definition of macro ‘BLAZE_STATIC_ASSERT’
  113 |    static_assert( expr, "Compile time condition violated" )
      |                   ^~~~
make[2]: *** [libWENOEXT/CMakeFiles/WENOEXT.dir/build.make:76: libWENOEXT/CMakeFiles/WENOEXT.dir/BlazeIO/BlazeIO.C.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:318: libWENOEXT/CMakeFiles/WENOEXT.dir/all] Error 2
make: *** [Makefile:136: all] Error 2


I appreciate your time and look forward to hear from you soon

**Comment by JanGaertner** [comment]
The Blaze cache size requirement could not be fulfilled. Where are you trying to install the library? You can set the BLAZE_CACHE_SIZE with:
```
./Allwmake -DBLAZE_CACHE_SIZE=<value>
```
See also here: https://bitbucket.org/blaze-lib/blaze/wiki/Configuration%20and%20Installation

**Comment by Tohid2020** [comment]
Thank you for replying. I am trying to install it for OpenFOAM 2012.  I am installing it on workbench pc with 16 cores. I am installing it in home directory.

**Comment by JanGaertner** [comment]
Try to set the Cache size to a useful value, e.g., 100000UL

**Comment by Tohid2020** [comment]
Thank you for your information and your replies.  Can I use weno schemes for compressible flows where there are discontinuities and high gradients?

**Comment by JanGaertner** [comment]
That is the main reason why the WENO scheme was built, yes.

**Comment by Tohid2020** [comment]
I appreciate your time and efforts.  The confusing part in the installation was :                                                                          
Supported OpenFOAM Versions:
OpenFOAM (ORG) v5.x - 8
OpenFOAM (ESI) v1912-v2012                                                                                                                                                                                                              While I installed it on OpenFOAM2312.   Thank you

**Comment by JanGaertner** [comment]
Indeed, it is fully compatible with OpenFOAM v2406. The dependencies on the OpenFOAM libraries are minimal and unlikely to change significantly. Thus, it should remain functional in future versions, unless major updates like those seen in OpenFOAM 9 occur.
